Resource.cs

```diff
@@ -0,0 +1,193 @@
+                return null;
+            }
+        }
+
+        public static async Task<Resource?> TestResource(Resource resource) {
+            if (!(resource.Provides ?? "").Split(",").Contains("server")) {
+                Console.WriteLine($"INFO: Skipping {resource.Name}/{resource.Product}, not a server");
+                return null;
+            }
+            foreach (Connection connection in resource.Connections ?? Enumerable.Empty<Connection>()) {
+                MainWindow.UserNameText += ".";
+                Uri Uri;
+                if (connection.Local)
+                    Uri = new UriBuilder("http", connection.Address, connection.Port).Uri;
+                else if (!Config.Settings.LocalAddress)
+                    Uri = new UriBuilder(connection.Uri).Uri;
+                else
+                    continue;
+
+                if (Config.Settings.Skipped.Contains($"{Uri}")) {
+                    Console.WriteLine($"INFO: Skipped {Uri} due to previous HTTP error, remove from config.json to retry");
+                    continue;
+                }
+
+                try {
+                    Console.WriteLine($"INFO: Testing {(connection.Local ? 'L' : 'R')} {Uri}status/sessions?X-Plex-Token={resource.AccessToken?[..3]}...");
+                    HttpRequestMessage requestMessage = new(HttpMethod.Get, $"{Uri}status/sessions?X-Plex-Token={resource.AccessToken}");
+                    requestMessage.Headers.Add("Accept", "application/json");
+
+                    HttpResponseMessage sendResponse = await MainWindow.httpClient.SendAsync(requestMessage);
+                    sendResponse.EnsureSuccessStatusCode();
+                    Console.WriteLine($"INFO: Success {(connection.Local ? 'L' : 'R')} {Uri}status/sessions?X-Plex-Token={resource.AccessToken?[..3]}...");
+                    if (connection.Local)
+                        resource.LocalUri ??= Uri;
+                    else
+                        resource.Uri ??= Uri;
+                    if (resource.LocalUri != null && resource.Uri != null) {
+                        break;
+                    }
+                } catch (TaskCanceledException) {
+                    Console.WriteLine($"WARN: Timeout {(connection.Local ? 'L' : 'R')} {Uri}status/sessions?X-Plex-Token={resource.AccessToken?[..3]}...");
+                    // Unreachable server, skip for now
+                } catch (HttpRequestException e) {
+                    Console.WriteLine($"WARN: Unable to access {Uri}status/sessions?X-Plex-Token={resource.AccessToken?[..3]}: {e.Message}");
+                    Console.WriteLine($"INFO: Adding {Uri} to Skipped list in config.json");
+                    Config.Settings.Skipped.Add($"{Uri}");
+                } catch (Exception e) {
+                    Console.WriteLine($"WARN: Unable to get resource: {e.Message} {e.InnerException}");
+                }
+            }
+            if (resource.LocalUri == null && resource.Uri == null)
```

Comment:
i think people use `is null` rather than `== null` nowadays but idk

Reply:
Apparently == was slightly faster than is in older versions, but they now produce the same bytecode. The impact would be negligible either way, down to personal taste imo https://stackoverflow.com/a/40676671

Reply:
makes sense